src/test/java/pages/BasePage.java

```diff
@@ -19,6 +20,7 @@ public BasePage( WebDriver givenDriver ){
     driver = givenDriver;
     wait = new WebDriverWait(driver, Duration.ofSeconds(5));
     actions = new Actions(driver);
+    PageFactory.initElements(driver,this);
```

Comment:
Good Work! By defining the Page Factory in a base page class, you centralize the initialization logic and avoid duplicating code across multiple page classes. This promotes code consistency and reduces the chances of errors or inconsistencies in the element initialization process.